### PR TITLE
post api's completed

### PR DESCRIPTION
**Pull request by Haris-Siddiqui**

### CODE REVIEW

File: src/handlers/Post.js [hunk 1]
```diff
@@ -1,55 +1,65 @@
-
 const Response = require("./Response");
-const { Post: PostModel } = require("../models/postModel");
+const { User } = require("../models");
 const mongoose = require("mongoose");
-const AWS= require('aws-sdk');
+const { Post: PostModel } = require("../models/postModel");
```
Comment (i-abdul-raheem) [inline src/handlers/Post.js:4]: Why are we not importing it directly from ../models
We should not be using ../models/postModal

File: src/handlers/Post.js [hunk 2]
```diff
@@ -58,52 +68,103 @@ class Post extends Response {
...
+
+  //       const s3Response = await s3.upload(params).promise();
+  //       mediaUrls = {  // Set mediaUrls as an object if file is uploaded
+  //         public_id: s3Response.Key,
+  //         url: s3Response.Location,
+  //       };
+  //     }
+
+  //     const newPost = new PostModel({ text, user_id, mediaUrls });
+  //     const savedPost = await newPost.save();
```
Comment (i-abdul-raheem) [inline src/handlers/Post.js]: Remove these comments

File: src/handlers/Post.js [hunk 3]
```diff
@@ -58,52 +68,103 @@ class Post extends Response {
...
+  //     const { mediaUrls, text } = req.body;
+  //     const user_id = req.user.id; // Assuming user's ID is available in req.user
+
+  //     const newPost = new PostModel({
+  //       mediaUrls,
+  //       text,
+  //       user_id,
+  //     });
+
+  //     const savedPost = await newPost.save();
```
Comment (i-abdul-raheem) [inline src/handlers/Post.js]: remove comments

File: src/models/postModel.js [hunk 4]
```diff
@@ -2,19 +2,21 @@ const mongoose= require('mongoose');
...
     user_id:{
-        type :String
-    }
+        type :mongoose.Schema.Types.ObjectId,
+        ref:'User'
+    },
+    // shared_id:{
+    //     type:[{
+    //         type:mongoose.Schema.Types.ObjectId,
+    //         ref:'User'
```
Comment (i-abdul-raheem) [inline src/models/postModel.js]: remove comments

File: src/routes/posts.js [hunk 5]
```diff
@@ -1,13 +1,25 @@
...
+router.post('/:id', handler.createPost);
+module.exports={posts: router};
 
-router.post('/', upload.single('file'), handler.createPost);
 
-module.exports = { posts: router };
+// const express = require('express');
+// const router = require('express').Router();
+// const multer = require('multer');
+// const { Post } = require('../handlers/Post');
```
Comment (i-abdul-raheem) [inline src/routes/posts.js]: remove comments

File: src/routes/posts.js [hunk 6]
```diff
@@ -1,13 +1,25 @@
...
+const {Post}=require('../handlers/Post');
 
-const handler = new Post();
+const handler= new Post();
+router.get('/', handler.getAllPosts);
 
-const storage = multer.memoryStorage(); // Store files in memory
-const upload = multer({ storage: storage });
+router.delete('/:id', handler.deletePostById);
+router.post('/:id', handler.createPost);
```
Comment (i-abdul-raheem) [inline src/routes/posts.js]: why do we need an id parameter while creating a post???

**Review by i-abdul-raheem — CHANGES_REQUESTED** [review]

File: src/handlers/Post.js (hunk 2 above)
Comment (Haris-Siddiqui) [reply src/handlers/Post.js]: done

File: src/handlers/Post.js (hunk 1 above)
Comment (Haris-Siddiqui) [reply src/handlers/Post.js:4]: it was not working with this

File: src/handlers/Post.js (hunk 3 above)
Comment (Haris-Siddiqui) [reply src/handlers/Post.js]: done

File: src/models/postModel.js (hunk 4 above)
Comment (Haris-Siddiqui) [reply src/models/postModel.js]: done

File: src/routes/posts.js (hunk 5 above)
Comment (Haris-Siddiqui) [reply src/routes/posts.js]: done

File: src/handlers/Post.js (hunk 2 above)
Comment (Haris-Siddiqui) [reply src/handlers/Post.js]: done

File: src/routes/posts.js (hunk 6 above)
Comment (Haris-Siddiqui) [reply src/routes/posts.js]: i set it

**Review by console-dot — APPROVED** [review]

**Review by i-abdul-raheem — APPROVED** [review]

**Pull request merged**